bin/core/src/api/read/log_recording.rs

```diff
@@ -0,0 +1,298 @@
+use anyhow::{Context, anyhow};
+use database::mungos::{
+  by_id::find_one_by_id,
+  find::find_collect,
+  mongodb::{bson::doc, options::FindOptions},
+};
+use komodo_client::{
+  api::read::{GetLogRecording, GetRecordedLogs, ListLogRecordings, SearchRecordedLogs},
+  entities::{
+    log_recording::{LogRecording, LogRecordingListItem, LogRecordingStatus},
+    permission::PermissionLevel,
+    update::Log,
+    SearchCombinator,
+  },
+};
+use resolver_api::Resolve;
+
+use crate::{
+  helpers::query::get_user,
+  permission::get_check_permissions,
+  state::db_client,
+};
+
+use super::ReadArgs;
+
+impl Resolve<ReadArgs> for GetLogRecording {
+  async fn resolve(
+    self,
+    ReadArgs { user }: &ReadArgs,
+  ) -> serror::Result<LogRecording> {
+    let recording = find_one_by_id(&db_client().log_recordings, &self.id)
+      .await
+      .context("failed to find log recording")?
+      .context("log recording not found")?;
+
+    // Check permissions for the target resource based on type
+    let (variant, id) = recording.target.extract_variant_id();
+    match variant {
+      komodo_client::entities::ResourceTargetVariant::Server => {
+        get_check_permissions::<komodo_client::entities::server::Server>(
+          &id, user, PermissionLevel::Read.into()
+        ).await?;
+      }
+      komodo_client::entities::ResourceTargetVariant::Stack => {
+        get_check_permissions::<komodo_client::entities::stack::Stack>(
+          &id, user, PermissionLevel::Read.into()
+        ).await?;
+      }
+      komodo_client::entities::ResourceTargetVariant::Deployment => {
+        get_check_permissions::<komodo_client::entities::deployment::Deployment>(
+          &id, user, PermissionLevel::Read.into()
+        ).await?;
+      }
+      _ => return Err(anyhow!("Invalid target type for log recording").into()),
+    }
+
+    Ok(recording)
+  }
+}
+
+impl Resolve<ReadArgs> for ListLogRecordings {
+  async fn resolve(
+    self,
+    ReadArgs { user }: &ReadArgs,
+  ) -> serror::Result<Vec<LogRecordingListItem>> {
+    let mut filter = doc! {};
+
+    if let Some(target) = &self.target {
+      filter.insert("target.type", target.extract_variant_id().0.to_string());
+      filter.insert("target.id", target.extract_variant_id().1.clone());
+    }
+
+    if self.active_only {
+      filter.insert("status", LogRecordingStatus::Recording.to_string());
+    }
+
+    let recordings = find_collect(
+      &db_client().log_recordings,
+      filter,
+      None,
```

Comment:
I think collection may grow too much with time to load all into memory, so this API should be made paginated with the same pagination pattern as ListUpdates api.